src/multi_vector_simulator/E1_process_results.py

```diff
@@ -76,6 +76,109 @@
 ASSET_GROUPS_DEFINED_BY_OUTFLUX = [ENERGY_CONVERSION, ENERGY_PRODUCTION]
 
 
+def cut_below_pico(value, label):
+    r"""
+    Function trims results of oemof optimization to positive values and rounds to 0, if within a certain precision threshold (of -10^-6)
+
+    Oemof termination is dependent on the simulation settings of oemof solph. Thus, it can terminate the optimization if the results are with certain bounds, which can sometimes lead to negative decision variables (capacities, flows). Negative values do not make sense in this context. If the values are between -10^-6 and 0, we assume that they can be rounded to 0, as they result from the precision settings of the solver. In that case the value is overwritten for the futher post-processing. This should also avoid SOC timeseries with doubtful values outside of [0,1]. If any value is a higher negative value then the threshold, its value is not changed but a warning raised.
+    Similarily, if a positive devision variable is detected that has a value lower then the theshold, it is assumed that this only happends because of the solver settings, and the values below the theshold are rounded to 0.
+
+    Parameters
+    ----------
+    value: float or pd.Series
+        Decision variable determined by oemof
+
+    label: str
+        String to be mentioned in the debug messages
+
+    Returns
+    -------
+
+    value: float of pd.Series
+        Decision variable with rounded values in case that slight negative values or positive values were observed.
+
+    Notes
+    -----
+
+    Tested with:
+    - E1.test_cut_below_pico_scalar_value_below_0_larger_treshhold
+    - E1.test_cut_below_pico_scalar_value_below_0_smaller_treshhold
+    - E1.test_cut_below_pico_scalar_value_0
+    - E1.test_cut_below_pico_scalar_value_larger_0
+    - E1.test_cut_below_pico_scalar_value_larger_0_smaller_threshold
+    - E1.test_cut_below_pico_pd_Series_below_0_larger_treshhold
+    - E1.test_cut_below_pico_pd_Series_below_0_smaller_treshhold
+    - E1.test_cut_below_pico_pd_Series_0
+    - E1.test_cut_below_pico_pd_Series_larger_0
+    - E1.test_cut_below_pico_pd_Series_larger_0_smaller_treshold
+    """
+    treshhold = 10 ** (-6)
+    text_block_start = f"The value of {label} is below 0"
+    text_block_set_0 = f"Negative value (s) are smaller then {-treshhold}. This is likely a result of the termination/precision settings of the cbc solver. As the difference is marginal, the value will be set to 0. "
+    text_block_oemof = "This is so far below 0, that the value is not changed. All oemof decision variables should be positive so this needs to be investigated. "
+
+    # flows
+    if isinstance(value, pd.Series):
+        # Identifies any negative values. Decision variables should not have a negative value
```

Comment:
Not sure what you mean by "decision variables" - this function is only used to process the outputs from oemof simulations, right?